### PR TITLE
DG updates in activity management and blocking of future dates and datetime

### LINKED ISSUE #261: Block all future dates

### PR DESCRIPTION
**Pull request by AlWo223**
Closes #261

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -137,82 +137,98 @@ Verifying if there is an existence of a diet goal using an ArrayList takes O(n)
...
+1. `AthletiCLI`: Facilitates the mechanism. It captures the user input and initiates the parsing and execution.
+2. `Parser` (`Activity Parser`): Interprets the user input, generating both the appropriate command object and 
+   the activity instance.
+3. `AddActivityCommand`: Encapsulates the execution of the `add-activity` command, adding the activity to the data.
+4. `ActivityChanges`: Contains the arguments of the activity to be added. It is used to transfer the data from the 
+   parser to the activity in a modular way.
+5. `Activity`: Represents the activity to be added. It is a superclass for specific activity types like Run, Swim and 
+   Cycle.
+6. `Data`: manages the current state of the activity list.
+7. `ActivityList`: maintains the list of all activities added to the application.
```
Comment (skylee03) [inline docs/DeveloperGuide.md:157]:
```suggestion
6. `Data`: Manages the current state of the activity list.
7. `ActivityList`: Maintains the list of all activities added to the application.
```

**Review by nihalzp — APPROVED** [review]
LGTM!

File: src/main/java/athleticli/parser/Parser.java [hunk 2]
```diff
@@ -188,6 +188,9 @@ public static LocalDateTime parseDateTime(String datetime) throws AthletiExcepti
         LocalDateTime datetimeParsed;
         try {
             datetimeParsed = LocalDateTime.parse(datetime.replace(" ", "T"));
+            if (datetimeParsed.isAfter(LocalDateTime.now())) {
+                throw new AthletiException(Message.MESSAGE_DATE_FUTURE);
+            }
```
Comment (nihalzp) [inline src/main/java/athleticli/parser/Parser.java:193]: If possible, we should have one junit test for this function for future dates.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -137,82 +137,98 @@ Verifying if there is an existence of a diet goal using an ArrayList takes O(n)
...
-One of the key data component in the parsing process is the `ActivityChanges` object. It is used for storing the 
-different attributes of the activity that are to be added. Later, the `ActivityParser` will use the `ActivityChanges`
-to create the `Activity` object. 
+The `ActivityChanges` object plays a key role in the parsing process. It is used for storing the 
+different attributes of the activity that are to be added. Later, the `ActivityParser` 
+will use the `ActivityChanges` to create the `Activity` object. 
 > This way of transferring data between the parser and the activity is more flexible which is suitable for future 
-extensions of the activity types and allows a more modular design. This design and most of the methods can be reused 
-for the `edit-activity` mechanism, which works in the same way with slight modifications due to optional parameters.
+extensions of the activity types and allows for a more modular design. This design and most of the methods can be
```
Comment (skylee03) [inline docs/DeveloperGuide.md:188]:
```suggestion
> extensions of the activity types and allows for a more modular design. This design and most of the methods can be 
```

File: src/main/java/athleticli/parser/Parser.java [hunk 4]
```diff
@@ -196,7 +199,11 @@ public static LocalDateTime parseDateTime(String datetime) throws AthletiExcepti
 
     public static LocalDate parseDate(String date) throws AthletiException {
         try {
-            return LocalDate.parse(date);
+            LocalDate dateParsed = LocalDate.parse(date);
+            if (dateParsed.isAfter(LocalDate.now())) {
+                throw new AthletiException(Message.MESSAGE_DATE_FUTURE);
+            }
```
Comment (nihalzp) [inline src/main/java/athleticli/parser/Parser.java:205]: Same as before

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -137,82 +137,98 @@ Verifying if there is an existence of a diet goal using an ArrayList takes O(n)
...
 </p>
 
-There exist three types of specific activities that inherit from the 'Activity' class: Run, Swim and Cycle. Each of 
-these classes has their own attributes and methods. The 'ActivityList' contains a list of all the activity instances.
+> The diagram shows the inheritance relationship between the `Activity` class and the specific activity types Run, 
+> Swim and Cycle, each with unique attributes and methods. This design becomes especially crucial in future 
+> development cycles with added parameters and activity types. The 'ActivityList' aggregates these instances.
 
-Given below is an example usage scenario and how the add mechanism behaves at each step.
+Usage Scenario and Process flow:
```
Comment (skylee03) [inline docs/DeveloperGuide.md:173]:
```suggestion
Usage Scenario and Process Flow:
```

**Pull request merged**